diag_manager/fms_diag_file_object.F90

```diff
@@ -745,9 +745,12 @@ end subroutine add_axes
 !> @brief adds the start time to the fileobj
 !! @note This should be called from the register field calls. It can be called multiple times (one for each variable)
 !! So it needs to make sure that the start_time is the same for each variable. The initial value is the base_time
-subroutine add_start_time(this, start_time)
-  class(fmsDiagFile_type), intent(inout)       :: this            !< The file object
+subroutine add_start_time(this, start_time, model_time)
+  class(fmsDiagFile_type), intent(inout)       :: this           !< The file object
   TYPE(time_type),         intent(in)          :: start_time     !< Start time to add to the fileobj
+  TYPE(time_type),         intent(out)         :: model_time     !< The current model time
```

Comment:
This is he start time.  Where does it become the current model time?

Reply:
This is being called from the `register_fields` calls (the beginning of the run)
so the `current_model_time` is the `start_time`

Reply:
it then gets updated after every diag_send_complete